### PR TITLE
refactor(ocrvs-12448): deprecate v1 name schema

### LINKED ISSUE #12448: Set firstname and surname mandatory in user table
In database, currently both are optional. Seemingly this was done for legacy reasons, but should not be needed anymore.

```
 const name: IUserName[] = input.name ?? [
    {
      use: 'en',
      given: [existingUser.firstname ?? ''],
      family: existingUser.surname ?? ''
    }
  ]
```

- [ ] Write migration for `users` table
- [ ] Update application code and clean up the conditionals.

### PR DESCRIPTION
**Pull request by Zangetsu101**
## Description
- firstname and surname are now mandatory in user schema
- Remove the conversion of v2 name -> v1 name then back again to v2 from client, events and commons.

I've left user-mgnt as is because it will get removed eventually

## Checklist

- [x] I have linked the correct Github issue under "Development"
- [x] I have tested the changes locally, and written appropriate tests
- [x] I have tested beyond the happy path (e.g. edge cases, failure paths)
- [ ] I have updated the changelog with this change (if applicable)
- [x] I have updated the GitHub issue status accordingly

### CODE REVIEW

File: packages/migration/src/migrations/events/1764251733685_create-users-table.sql
Comment (Zangetsu101) [inline packages/migration/src/migrations/events/1764251733685_create-users-table.sql:1]: Although not ideal, I went with updating the existing migration as it would otherwise require creating 3 more migrations across events and migrate-legacy-data.

File: packages/gateway/src/config/routes.ts [hunk 1]
```diff
@@ -44,23 +40,6 @@ export const getRoutes = () => {
...
-          'Generate a 6 digit verification code.' +
-          'Sends an SMS/email to the user with verification code.',
-        validate: {
-          payload: requestSchema
-        },
-        response: {
-          schema: responseSchema
-        }
-      }
-    },
```
Comment (Zangetsu101) [inline packages/gateway/src/config/routes.ts:63]: This is not used anymore from client due to switching to the user trpc route

**Review by rikukissa — APPROVED** [review]
